### PR TITLE
Change SwiftUI Image extension for loading ImageAssets

### PR DESCRIPTION
**Pull request by osanoj**
# Why?
I think the previous fix was not optimal, Always adding `ImageAsset` every time you load an image in SwiftUI feels a bit off. I think aligning with the UIKit extension we have is a better choice. This still solves issues if you import this package in a project that has Xcode asset generation turned on.

# What?
- Change our Image extension to `init(named assetName: ImageAsset)` - so you use it like `Image(named: .remove)` In the same way you would do `UIImage(named: .remove)`.
- Also make the Demo project compile if Xcode 15 asset generation is turned on.

# Version Change
Patch

### CODE REVIEW

**Review by ag12 — APPROVED** [review]
🧑‍🚒

File: FinniversKit/Sources/Fullscreen/LoadingView/LoadingSwiftUIView.swift [hunk 1]
```diff
@@ -38,7 +38,7 @@ public struct LoadingSwiftUIView: View {
                 // Since the indicator view starts at a smaller size we must wrap it for fixed size
                 VStack {
                     if showSuccess {
-                        Image(ImageAsset.checkmarkBig)
+                        Image(named:.checkmarkBig)
```
Comment (BjornRuud) [inline FinniversKit/Sources/Fullscreen/LoadingView/LoadingSwiftUIView.swift:41]: Should be space after label.

File: FinniversKit/Sources/SwiftUI Components/ResultSwiftUIView.swift [hunk 2]
```diff
@@ -64,7 +64,7 @@ struct ResultSwiftUIView_Previews: PreviewProvider {
     static var previews: some View {
         Group {
             ResultSwiftUIView(
-                image: Image(ImageAsset.ratingFaceDissatisfied),
+                image: Image(named:.ratingFaceDissatisfied),
```
Comment (BjornRuud) [inline FinniversKit/Sources/SwiftUI Components/ResultSwiftUIView.swift:67]: And here.

File: FinniversKit/Sources/SwiftUI Components/ResultSwiftUIView.swift [hunk 3]
```diff
@@ -74,7 +74,7 @@ struct ResultSwiftUIView_Previews: PreviewProvider {
             ).previewDisplayName("Error retry")
 
             ResultSwiftUIView(
-                image: Image(ImageAsset.magnifyingGlass),
+                image: Image(named:.magnifyingGlass),
```
Comment (BjornRuud) [inline FinniversKit/Sources/SwiftUI Components/ResultSwiftUIView.swift:77]: And here.

**Review by BjornRuud — APPROVED** [review]
Improved!

File: FinniversKit/Sources/Fullscreen/LoadingView/LoadingSwiftUIView.swift (hunk 1 above)
Comment (osanoj) [reply FinniversKit/Sources/Fullscreen/LoadingView/LoadingSwiftUIView.swift:41]: 🙈

Comment (ag12) [reply FinniversKit/Sources/Fullscreen/LoadingView/LoadingSwiftUIView.swift]: "God is in the details"

**Comment by ag12** [comment]
https://en.wikipedia.org/wiki/The_devil_is_in_the_details

while waiting for CI to finish.

**Pull request merged**